test/suite/test_timestamp22.py

```diff
@@ -0,0 +1,444 @@
+        # Returns: 1) whether it expects success, 2) config 3) new running commit timestamp
+        def timestamp_txn_config(commit_ts, running_commit_ts):
+            ok = True
+            config = ''
+            this_commit_ts = -1
+            if self.do_illegal():
+                # setting durable timestamp must be after prepare call
+                config += ',durable_timestamp=' + timestamp_str(self.gen_ts(commit_ts))
+                ok = False
+
+            # ODDITY: if we set the durable timestamp (which is illegal at this point), and set a
+            # valid commit timestamp, the timestamp_transaction() call will fail, but apparently,
+            # only the durable part fails.  The evidence is that the commit timestamp is set,
+            # as we get a complaint to that effect at the prepare call.  The issue is described
+            # in WT-6995.  This seems wrong, and it's hard to work around the problem, so we'll just
+            # avoid testing that situation for now.  Hence the check for a blank configuration.
+            if self.rand.rand32() % 2 == 0 and config == '':
+                if self.do_illegal():
+                    this_commit_ts = self.oldest_ts - 1
+                elif self.do_illegal():
+                    this_commit_ts = self.stable_ts - 1
+                else:
+                    # It's possible this will succeed, we'll check below.
+                    this_commit_ts = self.gen_ts(commit_ts)
+                config += ',commit_timestamp=' + timestamp_str(this_commit_ts)
+
+            if this_commit_ts >= 0:
+                if this_commit_ts < running_commit_ts:
+                    ok = False
+                if this_commit_ts < self.stable_ts:
+                    ok = False
+                if this_commit_ts < self.oldest_ts:
+                    ok = False
+            if not ok:
+                this_commit_ts = -1
+            if this_commit_ts >= 0:
+                running_commit_ts = this_commit_ts
+            return (ok, config, running_commit_ts)
+
+        session = self.session
+        needs_rollback = False
+        prepare_config = None
+        commit_config = 'commit_timestamp=' + timestamp_str(commit_ts)
+        tstxn1_config = ''
+        tstxn2_config = ''
+
+        ok_commit = do_prepare or not self.do_illegal()
+        ok_prepare = True
+        ok_tstxn1 = True
+        ok_tstxn2 = True
+
+        # Occasionally put a durable timestamp on a commit without a prepare,
+        # that will be an error.
+        if do_prepare or not ok_commit:
+            commit_config += ',durable_timestamp=' + timestamp_str(durable_ts)
+        cursor = session.open_cursor(self.uri)
+        prepare_ts = self.gen_ts(commit_ts)
+        prepare_config = 'prepare_timestamp=' + timestamp_str(prepare_ts)
+        begin_config = '' if read_ts < 0 else 'read_timestamp=' + timestamp_str(read_ts)
+
+        # We might do a timestamp_transaction calls either before/after inserting
```

Comment:
Should be `a timestamp_transaction call` (singular) or `might do timestamp_transaction calls`.